tests/ocpp_tests/test_sets/ocpp201/reservations.py

```diff
@@ -0,0 +1,1209 @@
+        test_utility,
+        charge_point_v201,
+        "ReservationStatusUpdate",
+        call_201.ReservationStatusUpdatePayload(
+            5, "Expired"
+        ),
+    )
+
+    # expect StatusNotification with status available
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "StatusNotification",
+        call_201.StatusNotificationPayload(
+            ANY, ConnectorStatusType.available, 1, 1
+        ),
+    )
+
+    # Try to start a transaction now on the previously reserved evse (which reservation is now expired).
+    await charge_point_v201.request_start_transaction_req(
+        id_token=IdTokenType(id_token=test_config.authorization_info.valid_id_tag_2,
+                             type=IdTokenTypeEnum.iso14443), remote_start_id=1, evse_id=1
+    )
+
+    # Now it succeeds.
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "RequestStartTransaction",
+        call_result201.RequestStartTransactionPayload(status="Accepted"),
+        validate_remote_start_stop_transaction,
+    )
+
+    # Start charging session
+    test_controller.plug_in()
+
+    # expect StatusNotification with status occupied
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "StatusNotification",
+        call_201.StatusNotificationPayload(
+            ANY, 'Occupied', 1, 1
+        ),
+    )
+
+    # expect TransactionEvent with event type 'Started'
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "TransactionEvent",
+        {"eventType": "Started"}
+    )
+
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "TransactionEvent",
+        {"eventType": "Updated"}
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.ocpp_version("ocpp2.0.1")
+async def test_reservation_connector_faulted(
+    test_config: OcppTestConfiguration,
+    charge_point_v201: ChargePoint201,
+    test_controller: TestController,
+    test_utility: TestUtility,
+):
+    """
+    Test if an evse can be reserved when the evse status is 'Faulted'
+    """
+    logging.info("######### test_reservation_connector_faulted #########")
+
+    # Set evse in state 'faulted'
+    test_controller.raise_error("MREC6UnderVoltage", 1)
+
+    await asyncio.sleep(1)
+
+    t = datetime.utcnow() + timedelta(minutes=10)
+
+    # Send a reserve new request
+    await charge_point_v201.reserve_now_req(
+        id=0,
+        id_token=IdTokenType(id_token=test_config.authorization_info.valid_id_tag_1,
+                             type=IdTokenTypeEnum.iso14443),
+        expiry_date_time=t.isoformat(),
+        evse_id=1
+    )
+
+    # which should return 'faulted'
+    assert await wait_for_and_validate(
+        test_utility,
+        charge_point_v201,
+        "ReserveNow",
+        call_result201.ReserveNowPayload(ReserveNowStatusType.faulted),
+    )
```

Comment:
also here you can do drop the sleep if you do

1. `raise_error`
2. wait and validate for StatusNotification.req(Faulted)
3. ReserveNow
4. wait and validate ReserveNow

Reply:
Same here:
We have a race condition here. The statusnotification is sent earlier than the status is received by the auth handler so the reservation handler does not know of the state of the connector yet. :(